src/services/services.ts

```diff
@@ -4051,6 +4054,90 @@ namespace ts {
                 log("getCompletionsAtPosition: getCompletionEntriesFromSymbols: " + (new Date().getTime() - start));
                 return uniqueNames;
             }
+
+            function getStringLiteralCompletionEntries(sourceFile: SourceFile, position: number) {
+                const node = findPrecedingToken(position, sourceFile);
+                if (!node || node.kind !== SyntaxKind.StringLiteral) {
+                    return undefined;
+                }
+
+                const argumentInfo = SignatureHelp.getContainingArgumentInfo(node, position, sourceFile);

```

Comment:
This seems like a slightly weird sort of dependency. Ideally there would be an API on the type checker like `getPossibleContextualTypes`